### ISSUE TITLE
url_sig location parameter is not created when a new DS is created with url signing keys

### ISSUE THREAD

**Issue by dneuman64** [apache/trafficcontrol#2919]
If you create a new delivery service with URL signing enabled, Traffic Ops does not create the url_sig location parameter.  This causes ORT to fail until the parameter is manually created and associated with Edge and Mid profiles. 

Traffic Ops should create the location parameter and associate is to profiles of servers that are assigned to the delivery service.

This used to work in the perl version of the code. 

I did not test URI signing keys, but I would assume that those do not work as well.

**Comment by rob05c** [comment]
TLDR Go is correctly emulating the Perl API, but the real bug is that the Perl API never did what the Perl UI does.

Looking at the Perl, it looks like it only created the `url_sig` `location` parameter on DS Update, in the UI; not the API, and not on Create.

That said, looking at the other location parameters in Perl, there are a couple of things here.
1. It looks like there's a mismatch between the UI and API. The Perl UI creates and assigns various location parameters, where the Perl API doesn't create some or all of them. 

2. The other location parameters appear to be created when the `deliveryservice_server` is created (though `url_sig` isn't, which appears to be a bug/oversight). Which makes sense: when the DS is first created, there are no servers assigned, so we can't know which Server Profiles to assign them to.

So, unless there are objections, my plan is to update the Go API to match the Perl UI, and create and assign all location params as necessary, in the `POST+PUT deliveryservice_server` endpoints rather than the `POST deliveryservice`.

That should make the Portal behave the same way the Perl UI does.

**Comment by alficles** [comment]
I can't speak fully for url sig, but URI Signing never created location parameters automatically. I thought we had a feature request opened for it, but I can't seem to find it.
